File: macosx/WallpaperInfoApp/WallpaperInfoApp/AppUtil.swift
//
//  WallpaperInfoUtil.swift
//  WallpaperInfoApp
//
//  Created by Brad Park on 4/13/20.
//  Copyright © 2020 Brad Park. All rights reserved.
//

import Foundation
import Cocoa

class MSG {
    public static var REQUEST_INFO:Int = 1;
    public static var PAUSE:Int = 2;
    public static var PREVIOUS:Int = 3;
    public static var NEXT:Int = 4;
    public static var SET_ROOT:Int = 5;
    public static var ACTIVITY_REPORT:Int = 6;
    public static var SET_WALLPAPER:Int = 7;
    public static var SET_INTERVAL:Int = 8;
    public static var SERVICE_INFO:Int = 9;
    public static var SET_THEME:Int = 10;
    public static var CUSTOM_CONFIG:Int = 11;
    public static var PREVIOUS_THEME:Int = 12;
    public static var NEXT_THEME:Int = 13;
	public static var TOGGLE_PAUSE:Int = 14;
	public static var SET_MODE:Int = 15;
	public static var SET_SAVER:Int = 16;
	public static var SET_SAVERTIME:Int = 17;
	public static var OPEN_SETTING_UI:Int = 18;
}

class AppUtil {
	
	static func getWordsArray(wordString:String) -> Array<String> {
		// check if empty string to avoid split function to create one empty entry in array
		if (wordString == "") {
			return Array<String>()
		} else {
			return wordString.components(separatedBy: "|")
		}
	}
    
	static func errorAlert(title:String, message:String) {
		let alert = NSAlert()
		alert.messageText = title
		alert.informativeText = message
		alert.alertStyle = .warning
		alert.addButton(withTitle: "OK")
		_ = alert.runModal()
	}
	
    static func isMyServiceRunning() -> Bool {
        let wpsi = WallpaperServiceInfo.getInstance();
        return wpsi.getWallpaperService().isStarted()
    }    
}

File: windows/WallpaperInfoApp/WallpaperInfoApp/AppUtil.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace WallpaperInfoApp {

	class MSG {
		public const int REQUEST_INFO = 1;
		public const int PAUSE = 2;
		public const int PREVIOUS = 3;
		public const int NEXT = 4;
		public const int SET_ROOT = 5;
		public const int ACTIVITY_REPORT = 6;
		public const int SET_WALLPAPER = 7;
		public const int SET_INTERVAL = 8;
		public const int SERVICE_INFO = 9;
		public const int SET_THEME = 10;
		public const int CUSTOM_CONFIG = 11;
		public const int PREVIOUS_THEME = 12;
		public const int NEXT_THEME = 13;
		public const int TOGGLE_PAUSE = 14;
	    public const int SET_MODE = 15;
	    public const int SET_SAVER = 16;
	    public const int SET_SAVERTIME = 17;
	    public const int OPEN_SETTING_UI = 18;
	}

	class AppUtil {

		public static List<String> getWordsArray(String wordString) {
			if (wordString.Equals("")) {
				return new List<String>();
			} else {
				return new List<String>(wordString.Split('|'));
			}
		}

	    public static bool isMyServiceRunning()  {
			WallpaperServiceInfo wpsi = WallpaperServiceInfo.getInstance();
			return wpsi.getWallpaperService().isStarted();
		}

		private static List<String> _allowedFilters = new List<String>();
	}
}
